Emby.Server.Implementations/Collections/CollectionManager.cs

```diff
@@ -344,7 +344,18 @@ public IEnumerable<BaseItem> CollapseItemsWithinBoxSets(IEnumerable<BaseItem> it
                     }
                     else
                     {
-                        results[item.Id] = item;
+                        var alreadyInResults = false;
+                        foreach (var child in item.GetMediaSources(true))
+                        {
+                            if (results.ContainsKey(Guid.Parse(child.Id)))
```

Comment:
The fact that the id is not already a guid makes me a bit wary. It might be safer to use `Guid.TryParse` here.

Reply:
I don't know why the id is a String, what should I do if the parse fails?

Reply:
skipping it seems like the best course of action